megamek/src/megamek/client/ui/swing/FiringDisplay.java

```diff
@@ -185,6 +238,42 @@ public FiringDisplay(final ClientGUI clientgui) {
         registerKeyCommands();
     }
 
+    @Override
+    protected void setButtons() {
+        buttons = new HashMap<>((int) (FiringCommand.values().length * 1.25 + 0.5));
+        for (FiringCommand cmd : FiringCommand.values()) {
+            String title = Messages.getString("FiringDisplay." + cmd.getCmd());
+            MegamekButton newButton = new MegamekButton(title, SkinSpecification.UIComponents.PhaseDisplayButton.getComp());
+            newButton.addActionListener(this);
+            newButton.setActionCommand(cmd.getCmd());
+            newButton.setEnabled(false);
+            buttons.put(cmd, newButton);
+        }
+        numButtonGroups = (int) Math.ceil((buttons.size() + 0.0) / buttonsPerGroup);
+    }
+
+    @Override
+    protected void setButtonsTooltips() {
```

Comment:
This is repeated a lot with minor differences. It looks like it could be unified as a static method to StatusBarPhaseDisplay.

Reply:
The few dependencies could be injected as parameters or obtained with overridden methods like getCommands()